packages/hash/frontend/src/pages/[account-slug]/types/new/entity-type.page.tsx

```diff
@@ -74,37 +75,49 @@ const Page: NextPageWithLayout = () => {
 
   const router = useRouter();
   const { authenticatedUser, loading } = useAuthenticatedUser();
+  const { orgs } = useOrgs();
 
   const { getEntityType } = useBlockProtocolGetEntityType();
 
-  if (
-    authenticatedUser &&
-    router.query["account-slug"] !== `@${authenticatedUser.shortname}`
-  ) {
-    void router.replace(
-      `/@${authenticatedUser.shortname}/types/new/entity-type`,
-    );
-  }
+  const shortname = router.query["account-slug"];
+
+  const namespace = useMemo(() => {
+    if (authenticatedUser && orgs) {
+      if (shortname === `@${authenticatedUser.shortname}`) {
+        return authenticatedUser.shortname;
+      }
+
+      const thisOrg = orgs?.find((org) => `@${org.shortname}` === shortname);
+
+      if (thisOrg) {
+        return thisOrg.shortname;
+      }
+
+      void router.replace(
```

Comment:
side effects must be in useEffect

Reply:
Addressed in 5413f143da89792a7695aff3013930c021b0de05.